include/deal.II/grid/construction_util.h

```diff
@@ -0,0 +1,120 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_grid_construction_utilities_h
+#define dealii_grid_construction_utilities_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/distributed/fully_distributed_tria.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A namespace dedicated to the struct ConstructionData, which can be used in
+ * Triangulation::create_triangulation().
+ */
+namespace construction
```

Comment:
This is my chance to rectify something that I think we got wrong last time: I find the name `ConstructionData` just a bit too generic, because it doesn't give away what is actually being constructed here. I missed that in the original PRs, but have rued the mistake ever since :-)

The same is true for the current namespace: `construction` in the global namespace could literally mean anything ;-) So let's be better about this then. How about `namespace GridDescription` (sticking with the various `Grid*` classes), or `namespace MeshDescription` (I think we also use the word "mesh" in some places), or just `namespace TriangulationDescription`.

The main class in the namespace would then simply be `Description` or again `XxxDescription`.

Reply:
I went with `TriangulationDescription::Description`. The corresponding files are called now `tria_description.h`/`tria_description.cc`.

The utility functions are called now:
* `TriangulationDescription::Utilities::create_description_from_triangulation`
* `TriangulationDescription::Utilities::create_description_from_triangulation_in_groups`